src/grokcore/annotation/tests/annotation/lazy.py

```diff
@@ -0,0 +1,87 @@
+"""
+  >>> grok.testing.grok(__name__)
+  >>> from zope import component
+  >>> from zope.annotation.attribute import AttributeAnnotations
+  >>> component.provideAdapter(AttributeAnnotations)
+  >>> manfred = Mammoth()
+
+We can adapt a model to an annotation interface and obtain a value for
+its attributes:
+
+  >>> lazyannotation = ILazy(manfred)
+  >>> lazyannotation.lazy_attribute
+  u'lazily waiting for a value.'
+
+Note how just "getting" the annotation attribute's default value (as no
+explicit value has been set before for this attribute), does not create
+a new annotation objects:
+
+  >>> print manfred.__dict__
+  {}
+
+If nothing was explicitly set before, we still can query for the annotation
+and indeed get None for it and it won't create an object:
+
+  >>> grok.queryAnnotation(manfred, ILazy) is None
+  True
+
+When setting an explicit value for the attribute on the annotations, we
+actually do store a new annotation object:
+
+  >>> lazyannotation.lazy_attribute = u'We have a value!'
+  >>> print manfred.__dict__
+  {'__annotations__': <...OOBTree object at ...>}
+
+  >>> print list(manfred.__dict__['__annotations__'].keys())
+  ['lazy.annotation.custom.name']
+
+Now the queryAnnotation will indeed return the annotations object:
+
+  >>> grok.queryAnnotation(manfred, ILazy)
+  <...Lazy object at ...>
+
+  >>> grok.queryAnnotation(manfred, ILazy).lazy_attribute
+  u'We have a value!'
+
+We can also delete the lazy annotation and all traces are gone:
```

Comment:
This is not actually true. It removes the annotation, but not the annotation storage. Since the key point of this change is to not initialize and write the storage until one actually writes a value *into* the storage, this wording might be a bit confusing.